connectors/src/connectors/confluence/index.ts

```diff
@@ -49,479 +49,516 @@ import { getAccessTokenFromNango } from "@connectors/lib/nango_helpers";
-      attributes: ["cloudId"],
-      where: {
-        connectorId,
-      },
-    });
-
-    const newConnection = await nango_client().getConnection(
-      getRequiredNangoConfluenceConnectorId(),
-      connectionId,
-      false
+    const confluenceCloudInformation = await getConfluenceCloudInformation(
+      confluenceAccessToken
     );
+    if (!confluenceCloudInformation) {
+      return new Err(new Error("Confluence access token is invalid"));
+    }
 
-    const confluenceAccessToken = newConnection?.credentials?.access_token;
-    const newConfluenceCloudInformation = await getConfluenceCloudInformation(
+    const userAccountId = await getConfluenceUserAccountId(
       confluenceAccessToken
     );
 
-    // Change connection only if "cloudId" matches.
-    if (
-      newConfluenceCloudInformation &&
-      currentCloudInformation &&
-      newConfluenceCloudInformation.id === currentCloudInformation.cloudId
-    ) {
-      await connector.update({ connectionId });
-
-      await nangoDeleteConnection(
-        oldConnectionId,
-        getRequiredNangoConfluenceConnectorId()
-      );
-    } else {
-      logger.info(
+    const { id: cloudId, url: cloudUrl } = confluenceCloudInformation;
+    try {
+      const confluenceConfigurationBlob = {
+        cloudId,
+        url: cloudUrl,
+        userAccountId,
+      };
+
+      const connector = await ConnectorResource.makeNew(
+        "confluence",
         {
-          connectorId,
-          newCloudId: newConfluenceCloudInformation?.id,
-          previousCloudId: currentCloudInformation?.cloudId,
+          connectionId: nangoConnectionId,
+          workspaceAPIKey: dataSourceConfig.workspaceAPIKey,
+          workspaceId: dataSourceConfig.workspaceId,
+          dataSourceName: dataSourceConfig.dataSourceName,
         },
-        "Cannot change the workspace of a Confluence connector"
+        confluenceConfigurationBlob
       );
 
-      // If the new connection does not grant us access to the same cloud id
-      // delete the Nango Connection.
-      await nangoDeleteConnection(
-        connectionId,
-        getRequiredNangoConfluenceConnectorId()
+      const workflowStarted = await launchConfluenceSyncWorkflow(
+        connector.id,
+        null
       );
+      if (workflowStarted.isErr()) {
+        return new Err(workflowStarted.error);
+      }
+
+      await launchConfluencePersonalDataReportingSchedule();
+
+      return new Ok(connector.id.toString());
+    } catch (e) {
+      logger.error({ error: e }, "Error creating confluence connector.");
+      return new Err(e as Error);
+    }
+  }
 
+  async update({
+    connectionId,
+  }: {
+    connectionId?: string | null;
+  }): Promise<Result<string, ConnectorsAPIError>> {
+    const connector = await ConnectorResource.fetchById(this.connectorId);
+    if (!connector) {
+      logger.error({ connectorId: this.connectorId }, "Connector not found.");
```

Comment:
Should this BaseConnectorManager exposes a this.log to abstract all those params?

Reply:
Same here, I dont want to change any logic in this PR, but happy t iterate